File: src/main/java/com/demonwav/mcdev/platform/mcp/at/psi/mixins/AtKeywordMixin.java
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2016 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.platform.mcp.at.psi.mixins;

import com.demonwav.mcdev.platform.mcp.at.AtElementFactory;

import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;

public interface AtKeywordMixin extends PsiElement {

    @NotNull
    PsiElement getKeywordElement();

    @NotNull
    AtElementFactory.Keyword getKeywordValue();

    void setKeyword(@NotNull AtElementFactory.Keyword keyword);
}

File: src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/psi/mixins/AtKeywordMixin.kt
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2019 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.platform.mcp.at.psi.mixins

import com.demonwav.mcdev.platform.mcp.at.AtElementFactory
import com.demonwav.mcdev.platform.mcp.at.psi.AtElement
import com.intellij.psi.PsiElement

interface AtKeywordMixin : AtElement {

    val keywordElement: PsiElement
    val keywordValue: AtElementFactory.Keyword

    fun setKeyword(keyword: AtElementFactory.Keyword)
}
